src/api/app/models/package.rb

```diff
@@ -196,7 +208,18 @@ def self.get_by_project_and_name(project_name_or_object, package_name, opts = {}
     project = internal_get_project(project_name_or_object)
     return unless project # remote prjs
 
-    return nil if project.scmsync.present?
+    if project.scmsync.present? && opts[:follow_project_scmsync_links]
```

Comment:
@hennevogel the intention is to keep the current behavior unless `opts[:follow_project_scmsync_links]` is set to true by hand right? at this place we used to check for `project.scmsync.present?` and returned `nil` otherwise. I think to make sure we still return `nil` in other cases it need a condition `return nil if project.scmsync.present? && !opts[:follow_project_scmsync_links]` after this block.